packages/devtools_app/integration_test/test_infra/run/_test_app_driver.dart

```diff
@@ -260,6 +183,154 @@ abstract class _TestApp with IOMixin {
     }
   }
 
+  Map<String, Object?>? _parseFlutterResponse(String line) {
+    if (line.startsWith('[') && line.endsWith(']')) {
+      try {
+        final Map<String, Object?>? resp = json.decode(line)[0];
+        return resp;
+      } catch (e) {
+        // Not valid JSON, so likely some other output that was surrounded by [brackets]
+        return null;
+      }
+    }
+    return null;
+  }
+}
+
+class TestDartCliApp extends IntegrationTestApp {
+  TestDartCliApp({
+    String appPath = 'test/test_infra/fixtures/empty_app.dart',
+  }) : super(appPath, TestAppDevice.cli);
+
+  static const vmServicePrefix = 'The Dart VM service is listening on ';
+
+  @override
+  Future<void> startProcess() async {
+    const separator = '/';
+    final parts = testAppPath.split(separator);
+    final scriptName = parts.removeLast();
+    final workingDir = parts.join(separator);
+    runProcess = await Process.start(
+      'dart',
+      [
+        '--observe=0',
+        'run',
+        scriptName,
+      ],
+      workingDirectory: workingDir,
+    );
+  }
+
+  @override
+  Future<void> waitForAppStart() async {
+    final vmServiceUri = await waitFor(
+      message: vmServicePrefix,
+      timeout: IntegrationTestApp._appStartTimeout,
+    );
+    final parsedVmServiceUri = Uri.parse(vmServiceUri);
+
+    // Map to WS URI.
+    _vmServiceWsUri =
+        convertToWebSocketUrl(serviceProtocolUrl: parsedVmServiceUri);
+  }
+
+  Future<String> waitFor({required String message, Duration? timeout}) {
+    final response = Completer<String>();
+    late StreamSubscription<String> sub;
```

Comment:
Maybe I'm misunderstanding, but why is this `late`?

Reply:
This is late so that we can reference `sub` in the listener callback.